Files/Helpers/Convert/StorageItemHelpers.cs

```diff
@@ -12,28 +12,46 @@ namespace Files.Helpers
     /// </summary>
     public static class StorageItemHelpers
     {
-        public static async Task<IStorageItem> ToStorageItem(this string path) 
+        public static async Task<IStorageItem> ToStorageItem(this string path) // TODO: If the TODO of IStorageItemWithPath is implemented, change return type to IStorageItem
         {
-            FilesystemResult<StorageFile> fsFileResult = await FilesystemTasks.Wrap(() => StorageFileExtensions.DangerousGetFileFromPathAsync(path));
+            StorageFolderWithPath rootFolder = await FilesystemTasks.Wrap(() => DrivesManager.GetRootFromPathAsync(path));
+
+            FilesystemResult<StorageFile> fsFileResult = await FilesystemTasks.Wrap(() => StorageFileExtensions.DangerousGetFileFromPathAsync(path, rootFolder));
 
             if (fsFileResult)
             {
-                return fsFileResult.Result;
+                if (!string.IsNullOrWhiteSpace(fsFileResult.Result.Path))
+                    return fsFileResult.Result;
+                else
+                {
+                    FilesystemResult<StorageFileWithPath> fsFileWithPathResult = await FilesystemTasks.Wrap(() => StorageFileExtensions.DangerousGetFileWithPathFromPathAsync(path, rootFolder));
+
+                    if (fsFileWithPathResult)
+                        return /* fsFileWithPathResult.Result */ null; // Could be done if IStorageItemWithPath implemented IStorageItem
```

Comment:
:confused:

Reply:
Is implementing IStorageItem a possibility. Or would it have to work different way?